pkg/azure/api/providerspec.go

```diff
@@ -278,3 +291,10 @@ type AzureDiagnosticsProfile struct {
 	// If not specified azure managed storage will be used.
 	StorageURI *string `json:"storageURI,omitempty"`
 }
+
+// AzureBetaFeatures is storing information about beta features.
+type AzureBetaFeatures struct {
```

Comment:
Is there reason to have beta features? I would recommend not to introduce this unless really required. I would keep the code simple. For your current requirement is it not sufficient to check if `SecurityProfile` has been set, if yes then skip the market place agreement. This can be easily abstracted into a function with proper doc strings so that the code is maintainable and understandable for other maintainers as well.

Reply:
> For your current requirement is it not sufficient to check if SecurityProfile has been set, if yes then skip the market place agreement.

Yes, but only because the currently available testing image for us is a SUSE preview image because that has a Gardener extension. There is a publicly available RedHad image but it is not "usable" by us at the moment. Hence, the `SkipMarketplaceAgreementAnnotation` is also there so that we do not commit a change to the MCM-azure spec that we would need to support in the long future (also explained in https://github.com/gardener/machine-controller-manager-provider-azure/pull/146#discussion_r1591920624). We don't skip the license check because we use a ConfidentialVM, but because the only testing image we have comes with limitations. So I don't think that checking the securityProfile should be gate for skipping the LA.

Reply:
I understood now that doing an agreement check should not be bound to `SecurityProfile`. As i understood from your above comment that going forward there will also be a possibility that the OS image that is used can have an associated license (which then needs to be checked). Therefore we should then code for both the possibilities ( no-license-check, license-check). This means that we could then possibly introduce a boolean allowing the consumer to set if a license check is required for the OS image used. The customer will have to know this (as you know it today for the test SUSE OS image).

Reply:
> his means that we could then possibly introduce a boolean allowing the consumer to set if a license check is required for the OS image used.

The boolean was actually the first implementation and arguably the easiest. It is however the only one that doesn't map to an actual VM attribute, which is what led to this change. I will revert it back then 👍